File: LeetCode/504_Base7/Solution.java
public class Solution {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }

        String result = "";
        int n = Math.abs(num);

        while (n != 0) {
            int r = n % 7;
            result = r + result;
            n /= 7;
        }

        return (num < 0 ? "-" : "") + result;
    }
}

File: LeetCode/504_Base7/solution.js
/**
 * @param {number} num
 * @return {string}
 */

const convertToBase7 = function(num) {
    if (!num) {
        return '0';
    }

    let result = '';
    let n = Math.abs(num);

    while (n) {
        let r = n % 7;
        result = r + result;
        n = Math.floor(n / 7);
    }

    return (num > 0 ? '' : '-') + result;
};
